services/TraineeNotificationsService.ts

```diff
@@ -6,16 +6,21 @@ export class TraineeNotificationsService extends ApiService {
   constructor() {
     super("/api");
   }
+  
   async getAllNotifications(
     params?: Record<string, string | number>
-  ): Promise<AxiosResponse<NotificationPage>> {    
-    const page = params?.page ? params?.page : "";
-    const size = params?.size ? params?.size : "";
-    const sort = params?.sort ? params?.sort : "";
-    const type = params?.type ? params?.type : "";
-    const status = params?.status ? params?.status : "";
-    const keyword = params?.keyword ? params?.keyword : "";
-    return this.get<NotificationPage>(`/notifications?page=${page}&size=${size}&sort=${sort}&type=${type}&status=${status}&keyword=${keyword}`);
+  ): Promise<AxiosResponse<NotificationPage>> {
+    const searchParams = new URLSearchParams();
+  
+    if (params) {
+      for (const [key, value] of Object.entries(params)) {
+        if (value !== undefined && value !== null) {
+          searchParams.append(key, String(value));
+        }
+      }
+    }
+  
+    return this.get<NotificationPage>(`/notifications?${searchParams.toString()}`);
```

Comment:
Can we have a condition where there are no searchParams (i.e. a blank string after `.toString()` ) or will we always call `/notifications` with at least one?

Reply:
I have add the condition to handle when searchParams is empty, and add some jest test for getAllNotifications(). I will add more tests in the follow-up PR when the notifications part is refactored.